app/tests/ble/central/src/main.c

```diff
@@ -0,0 +1,370 @@
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ble_central, 4);
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/sys/byteorder.h>
+
+#ifdef CONFIG_ARCH_POSIX
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "cmdline.h"
+#include "soc.h"
+
+static bool disconnect_and_reconnect = false;
+static bool clear_bond_on_disconnect = false;
+static bool halt_after_bonding = false;
+static int32_t wait_on_start = 0;
+
+static void ble_central_native_posix_options(void) {
+    static struct args_struct_t options[] = {
+        {.is_switch = true,
+         .option = "disconnect_and_reconnect",
+         .type = 'b',
+         .dest = (void *)&disconnect_and_reconnect,
+         .descript = "Disconnect and reconnect after the initial connection"},
+        {.is_switch = true,
+         .option = "halt_after_bonding",
+         .type = 'b',
+         .dest = (void *)&halt_after_bonding,
+         .descript = "Halt any further logic after bonding the first time"},
+        {.is_switch = true,
+         .option = "clear_bond_on_disconnect",
+         .type = 'b',
+         .dest = (void *)&clear_bond_on_disconnect,
+         .descript = "Clear bonds on disconnect and reconnect"},
+        {// .is_switch = true,
+         .option = "wait_on_start",
+         .name = "milliseconds",
+         .type = 'u',
+         .dest = (void *)&wait_on_start,
+         .descript = "Time in milliseconds to wait before starting the test process"},
+        ARG_TABLE_ENDMARKER};
+
+    native_add_command_line_opts(options);
+}
+
+NATIVE_TASK(ble_central_native_posix_options, PRE_BOOT_1, 1);
+// NATIVE_TASK(flash_native_posix_cleanup, ON_EXIT, 1);
+
+#endif
+
+static void start_scan(void);
+
+static struct bt_conn *default_conn;
+
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+static struct bt_gatt_discover_params discover_params;
+static struct bt_gatt_subscribe_params subscribe_params;
+
+static uint8_t notify_func(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
+                           const void *data, uint16_t length) {
+    if (!data) {
+        LOG_DBG("[UNSUBSCRIBED]");
+        params->value_handle = 0U;
+        return BT_GATT_ITER_STOP;
+    }
+
+    LOG_HEXDUMP_DBG(data, length, "payload");
+
+    return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+                             struct bt_gatt_discover_params *params) {
+    int err;
+
+    if (!attr) {
+        LOG_DBG("Discover complete");
```

Comment:
Many of the other messages are surrounded with `[]`, which I assume is to make them easier to pick out of logs. However, some of them like this one aren't. Should all of them be formatted like that, or is there a difference between the two types of messages?

Reply:
No difference... I'll update them for consistency, but it's not really of significance.

Reply:
Modified to be more consistent.